src/poetry/packages/locker.py

```diff
@@ -241,6 +250,7 @@ def set_lock_data(self, root: Package, packages: list[Package]) -> bool:
             "lock-version": self._VERSION,
             "python-versions": root.python_versions,
             "content-hash": self._content_hash,
+            # TODO stop writing files here, this is deprecated.
```

Comment:
If we bump the `lock-version`, there is no reason to keep writing files here, isn't it?

Reply:
Depends whether we're ready to make a breaking change or not.

If we stop writing files here then the version should go 2.0, so that down-level poetry knows that it can't read it.  

I was expecting that we'd go 1.2 now, and 2.0 at some later point, but am happy to be told otherwise.

Reply:
Ah ok, I thought that any version bump makes it unreadable for older poetry. (I didn't look into the code.)

I have concerns that the lockfile size will increase significantly if we write the information twice. It seems we can only choose between the devil and the deep blue sea... I think we have to discuss that among maintainers.